src/MemoryProtocol.js

```diff
@@ -0,0 +1,129 @@
+import invariant from 'invariant';
+
+import createPath from './utils/createPath';
+import ensureLocation from './utils/ensureLocation';
+
+const STATE_KEY = '@@farce/state';
+
+export default class MemoryProtocol {
+  constructor(initialLocation, { persistent = false } = {}) {
+    this._persistent = persistent;
+
+    const initialState = persistent ? this._loadState() : null;
+    if (initialState) {
+      this._stack = initialState.stack;
+      this._index = initialState.index;
+    } else {
+      this._stack = [ensureLocation(initialLocation)];
+      this._index = 0;
+    }
+
+    this._keyPrefix = Math.random()
+      .toString(36)
+      .slice(2, 8);
+    this._keyIndex = 0;
+
+    this._listener = null;
+  }
+
+  _loadState() {
+    try {
+      const { stack, index } = JSON.parse(
+        window.sessionStorage.getItem(STATE_KEY),
+      );
+
+      // Checking instanceof Array is okay because we make the array.
+      if (
+        stack instanceof Array &&
```

Comment:
This is still weird :P what's wrong with isArray?

Reply:
requires a polyfill for older browsers, no? i don't have babel-runtime here... not sure what the right thing is

Reply:
how far back are you going? `isArray` is ie9+